R/req-body.R

```diff
@@ -216,23 +221,28 @@ req_body_apply <- function(req) {
   } else if (type == "raw") {
     req <- req_body_apply_raw(req, data)
   } else if (type == "json") {
-    content_type <- "application/json"
     json <- exec(jsonlite::toJSON, data, !!!req$body$params)
     req <- req_body_apply_raw(req, json)
   } else if (type == "multipart") {
     data <- unobfuscate(data)
-    content_type <- NULL
     req$fields <- data
   } else if (type == "form") {
     data <- unobfuscate(data)
-    content_type <- "application/x-www-form-urlencoded"
     req <- req_body_apply_raw(req, query_build(data))
   } else {
     abort("Unsupported request body `type`", .internal = TRUE)
   }
 
-  # Must set header afterwards
+  # Respect existing Content-Type if set
+  type_idx <- match("content-type", tolower(names(req$headers)))
```

Comment:
I've changed the logic a bit here — now if you manually override with the `content-type` header, we don't check it. I think this is ok, and gives the user an escape hatch if they're having problems. But generally we expect people to set the content type via the appropriate `req_body_*` function, which will verify that the type is as expected.